config/routes.rb

```diff
@@ -2622,17 +2622,16 @@
   get '/pictures/:basename' => 'picture#show', :basename => /[\da-zA-Z]+\.[\da-zA-Z]+/
 
   # Enablement for the REST API
-  # OPTIONS requests for REST API pre-flight checks
-
   # Semantic Versioning Regex for API, i.e. vMajor.minor.patch[-pre]
   apiver_regex = /v[\d]+(\.[\da-zA-Z]+)*(\-[\da-zA-Z]+)?/
 
-  match '/api/*path'   => 'api#handle_options_request', :via => [:options]
-  get '/api(/:version)'           => 'api#show',    :format => 'json', :version => apiver_regex
+  # OPTIONS requests for REST API pre-flight checks
+  match '/api/*path' => 'api#handle_options_request', :via => [:options]
 
-  get    '/api(/:version)/:collection(/:c_id(/:subcollection(/:s_id)))' => 'api#show',    :format => 'json', :version => apiver_regex
-  match  '/api(/:version)/:collection(/:c_id(/:subcollection(/:s_id)))' => 'api#update',  :format => 'json', :via => [:post, :put, :patch], :version => apiver_regex
-  delete '/api(/:version)/:collection(/:c_id(/:subcollection(/:s_id)))' => 'api#destroy', :format => 'json', :version => apiver_regex
+  get    '/api(/:version)(/*c_path)' => 'api#show',    :format => 'json', :version => apiver_regex
```

Comment:
not really changing the routing here (it's all /api), just moving the parsing down to our RequestAdapter. We could extend the routes.rb to have the settings (and soon automate) entries declared before the above for now, but I've been thinking about the virtual subcollections in the future (i.e. being to specify /api/vms/:id/storage/disks/:id/...  all based on reflections/virtual attributes/etc.) where our current collection/c_id/subcollection/s_id layout would fail. It would be easier to support virtual subcollections/resources if parsing is done on our side.